goGene/fromGtf.go

```diff
@@ -0,0 +1,122 @@
+package goGene
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/fasta"
+	"github.com/vertgenlab/gonomics/gtf"
+)
+
+// GtfToGoGene converts a gtf record into a GoGene data structure
+// WARNING: If multiple isoforms are present, only the longest is used
+func GtfToGoGene(g *gtf.Gene, ref []*fasta.Fasta) *GoGene {
+	answer := new(GoGene)
+	gtf.MoveCanonicalToZero(g)
+	transcript := g.Transcripts[0]
+
+	answer.id = g.GeneID
+	answer.strand = transcript.Strand
+
+	if transcript.Strand { // If on the positive strand
+		answer.startPos = transcript.Start - 1
+
+		fastaMap := fasta.FastaMap(ref)
+		answer.genomeSeq = make([]dna.Base, transcript.End-(transcript.Start-1))
+		copy(answer.genomeSeq, fastaMap[transcript.Chr][transcript.Start-1:transcript.End])
+		answer.cdnaSeq = make([]dna.Base, 0, gtf.CdsLength(transcript))
+		answer.featureArray = make([]Feature, len(answer.genomeSeq))
+		answer.cdsStarts = make([]int, 0, len(transcript.Exons))
+		answer.cdsEnds = make([]int, 0, len(transcript.Exons))
+
+		var i int
+		var prevExonEnd int = answer.startPos
+		var currCdsPos int32
+		for _, val := range transcript.Exons {
+			for i = prevExonEnd - answer.startPos; i < val.Start-1-answer.startPos; i++ {
+				answer.featureArray[i] = Intron
```

Comment:
Do we need to make sure that there are no overlapping genes in the gtf?  It seems like otherwise later gtf records overwrite earlier ones.

Reply:
Not sure I'm following, couldnt all this information be scattered to different lines of a gtf file?

Reply:
Actually i think i remember vaguely there was some functionality that logically handles this, is that what `gtf.Gene` is?

Reply:
Yes. Gtf.Gene holds all the same information. The goal here is to make that information more accessible. If I gave you a gtf.Gene and asked what is the cDNA position for chr3:45877890 you would need to write a block of code to get that answer. Here you just need to do featureArray[45877890]. It seems like overkill at first. But it greatly simplifies complex tasks like predicting the effect of an insertion or deletion.